File: Backend/src/models/organisation_linked_sites.py
# Backend/src/models/organisation_linked_sites.py
# Definition of the OrganisationLinkedSite model, connecting Organisations and linked sites

# Author: Valentin Haas, 2024

# Library imports
from sqlalchemy import Column, Integer

# Project imports
from ._base import BaseTable


class OrganisationLinkedSitesTable(BaseTable):
    """Definition of the OrganisationLinkedSite model for the database."""

    __tablename__ = "OrganisationLinkedSites"

    Organisation = Column(Integer, primary_key=True, foreign_key="Organisations.Id")
    """The unique identifier of the Organisation."""

    Site = Column(Integer, primary_key=True, foreign_key="LinkedSites.Id")
    """The unique identifier of the linked site."""

File: Backend/src/models/organisationlinkedsites.go
// Backend/src/models/organisationlinkedsites.go
// Definition of the OrganisationLinkedSite model, connecting Organisations and linked sites

// Author: Valentin Haas, 2025
package models

import (
	"gorm.io/gorm"
)

// OrganisationLinkedSites is a model to connect Organisations and linked sites.
type OrganisationLinkedSite struct {
	gorm.Model                  // Provides ID, CreatedAt, UpdatedAt, DeletedAt fields
	Organisation   Organisation // Referenced Organisation Object
	OrganisationID uint         `gorm:"primaryKey"` // The unique identifier of the Organisation.
	Site           LinkedSite   // Referenced LinkedSite Object
	SiteID         uint         `gorm:"primaryKey"` // The unique identifier of the linked site.
}
